jinahub/indexers/storage/MongoDBStorage/__init__.py

```diff
@@ -0,0 +1,126 @@
+    new_doc = Document(d, copy=True, hash_content=False)
+    new_doc.ClearField('embedding')
+    return new_doc.SerializeToString()
+
+
+class MongoDBStorage(Executor):
+    def __init__(
+        self,
+        host: str = '127.0.0.1',
+        port: int = 27017,
+        username: Optional[str] = '',
+        password: Optional[str] = '',
+        database: str = 'jina_index',
+        collection: str = 'jina_index',
+        default_traversal_paths: List[str] = ['r'],
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self._handler = MongoHandler(
+            host=host,
+            port=port,
+            username=username,
+            password=password,
+            database=database,
+            collection=collection,
+        )
+        self._traversal_paths = default_traversal_paths
+
+    @requests(on='/index')
+    def add(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """Add Documents to MongoDB
+
+        :param docs: list of Documents
+        :param parameters: parameters to the request
+        """
+        traversal_paths = parameters.get('traversal_paths', self._traversal_paths)
+        self._handler.add(docs.traverse_flat(traversal_paths))
+
+    @requests(on='/update')
+    def update(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """Updated document from the database.
+
+        :param docs: list of Documents
+        :param parameters: parameters to the request
+        """
+        traversal_paths = parameters.get('traversal_paths', self._traversal_paths)
+        self._handler.update(docs.traverse_flat(traversal_paths))
+
+    @requests(on='/delete')
+    def delete(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """Delete document from the database.
+
+        :param docs: list of Documents
+        :param parameters: parameters to the request
+        """
+        traversal_paths = parameters.get('traversal_paths', self._traversal_paths)
+        self._handler.delete(docs.traverse_flat(traversal_paths))
+
+    @requests(on='/search')
+    def search(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """Get the Documents by the ids of the docs in the DocArray
+
+        :param docs: the DocumentArray to search with (they only need to have the `.id` set)
+        :param parameters: the parameters to this request
+        """
+        traversal_paths = parameters.get('traversal_paths', self._traversal_paths)
+        self._handler.search(docs.traverse_flat(traversal_paths))
+
+    @requests(on='/dump')
+    def dump(self, parameters: Dict = {}, **kwargs):
+        """Dump the index
+
+        :param parameters: a dictionary containing the parameters for the dump
+        """
+
+        path = parameters.get('dump_path')
+        if path is None:
+            self.logger.error(f'No "dump_path" provided for {self}')
+
+        shards = int(parameters.get('shards'))
```

Comment:
This will error out if `shards` is not set.

Reply:
Also, you log an error - but then what? The code in your case proceeds, so what will happen, will `export_dump_streaming` error out? I think this is bad UX.

Reply:
Consider raising an error instead, and then add tests for this